src/libraries/System.Text.RegularExpressions/tests/Regex.MultipleMatches.Tests.cs

```diff
@@ -140,6 +140,19 @@ public static IEnumerable<object[]> Matches_TestData()
                     new CaptureData("C789", 10, 4),
                 }
             };
+
+            yield return new object[]
```

Comment:
since you didn't have to modify an existing test to _not_ run when `RegexOptions.ECMAScript` is set, we must be missing a test that verifies that `[^]` throws when `RegexOptions.ECMAScript` is not set.